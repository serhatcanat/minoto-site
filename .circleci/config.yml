# Javascript Node CircleCI 2.0 configuration file
#
# Check https://circleci.com/docs/2.0/language-javascript/ for more details
#
version: 2.1
orbs:
  aws-s3: circleci/aws-s3@1.0.11
  cypress: cypress-io/cypress@1
jobs:
  build:
    docker:
      - image: circleci/node:10.12-browsers

    steps:
      - checkout

      # Download and cache dependencies
      - restore_cache:
          name: Restore Yarn Package Cache 4
          keys:
            - yarn-packages-{{ checksum "yarn.lock" }}
      - run:
          name: Install Dependencies
          command: yarn install --frozen-lockfile
      - save_cache:
          name: Save Yarn Package Cache
          key: yarn-packages-{{ checksum "yarn.lock" }}
          paths:
            - ~/.cache/yarn

      # run build!

      - run: yarn build
      # run linter!
      - run: yarn lint
      # run tests!
      - run: yarn test

      - run: yarn ci:cypress-run

      - run: npx semantic-release

      - run:
          name: SETTING_ENV
          command: |
            if [ $CIRCLE_BRANCH = "master" ]; then
            echo 'export S3_BUCKET="s3://dev-minoto-web/"' >> $BASH_ENV
            else
            echo 'export S3_BUCKET="s3://staging-minoto-web/"' >> $BASH_ENV
            fi

      - aws-s3/sync:
          from: build
          to: $S3_BUCKET
          arguments: |
            --acl public-read \
            --cache-control "max-age=86400"
          overwrite: true

