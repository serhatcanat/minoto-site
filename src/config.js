let domain = window.location.hostname

let endpoint = ''

switch (domain) {
    case 'localhost':
        endpoint = 'http://localhost:8000'
        break;
    case 'minoto.thinkerfox.com':
        endpoint = 'http://minoto-api.thinkerfox.com'
        break;
    default:
        endpoint = 'https://api.minoto.com'
        break;
}

export const apiBase = `${endpoint}/v1/shared/`

export const storagePath = 'https://minoto-test.ams3.digitaloceanspaces.com/'
export const basePath = 'https://minoto.com'