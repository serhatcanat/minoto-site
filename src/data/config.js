module.exports = {
	postTitle: 'Minoto',
	mobileBreakPoint: 960,
	mapsAPIkey: "AIzaSyDncyEm2ApXXZ0x8D8QNdqtj8S3j2Sn-4M",
	mapsTheme: false,
	gClient: "129152301361-mtuvr3rnnrbs9774mpp0ej7r4ta4r2hr.apps.googleusercontent.com",
	fClient: "2469148543124127",
	filterSeperator: "|",

	social: {
		instagramURL: "http://www.instagram.com/minotocom",
		youtubeURL: "https://www.youtube.com/minotocom",
		facebookURL: "http://www.facebook.com/minotocom",
		twitterURL: "http://www.twitter.com/minotocom",
		linkedinURL: "https://www.linkedin.com/company/minoto",
	}
}