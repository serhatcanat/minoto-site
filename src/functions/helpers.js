import extend from "lodash/extend"
import { storagePath, apiBase } from "../config"
import { disableBodyScroll, enableBodyScroll } from 'body-scroll-lock';

export function scrollTo(inputOpts, endFunction = false) {
	let defaultOpts = {
		to: 0,
		offset: 0,
		duration: 'auto',
		easing: true,
		target: 'body',
		axis: 'y',
	}

	let opts = extend({}, defaultOpts, inputOpts);

	if (!isDefined(window.scrollTimeout)) {
		window.scrollTimeout = {};
	}

	let id = 'body';

	if (opts.target !== 'body') {
		if (!opts.target.hasAttribute('id')) {
			opts.target.id = uid();
		}

		id = opts.target.id;
	}

	if (!isDefined(window.scrollTimeout[id])) {
		window.scrollTimeout[id] = null;
	}

	let to = opts.to;
	let offset = opts.offset;
	let duration = opts.duration;
	let easing = opts.easing;

	let axisString = (opts.axis === 'y' ? 'Top' : 'Left');
	//let sizeString = (opts.axis === 'y' ? 'Height' : 'Width');

	if (to !== parseFloat(to, 10)) {
		if (opts.target === 'body') {
			to = to.getBoundingClientRect()[axisString.toLowerCase()] + document.documentElement['scroll' + axisString];
		} else {
			to = to['offset' + axisString];
		}
		//to = to.offsetTop + window.innerHeight;
	}

	to = to + offset;

	if (duration === 0) {
		if (opts.target === 'body') {
			document.documentElement['scroll' + axisString] = to;
			document.body['scroll' + axisString] = to;
		} else {
			opts.target['scroll' + axisString] = to;
		}
	} else {
		let start = (opts.target === 'body' ? Math.max(document.body['scroll' + axisString], document.documentElement['scroll' + axisString]) : opts.target['scroll' + axisString]),
			change = to - start,
			currentTime = 0,
			increment = 20;

		if (duration === 'auto') {
			let distance = Math.abs(start - to);

			let multiplier = .8;

			if (distance < 500) {
				multiplier = 1.2;
			}
			if (distance > 1500) {
				multiplier = .5;
			}
			duration = distance * multiplier;

			if (duration < 500) { duration = 500; }
			if (duration > 900) { duration = 900; }
		}
		var animateScroll = function() {
			if (window.scrollTimeout[id] !== null) {
				clearTimeout(window.scrollTimeout[id]);
				window.scrollTimeout[id] = null;
			}
			currentTime += increment;
			//console.log(currentTime);
			var val = 0;
			if (easing) {
				val = Math.easeInOutQuad(currentTime, start, change, duration);
			} else {
				val = Math.linearTween(currentTime, start, change, duration);
			}
			if (opts.target === 'body') {
				document.documentElement['scroll' + axisString] = val;
				document.body['scroll' + axisString] = val;
			} else {
				opts.target['scroll' + axisString] = val;
			}

			if (currentTime < duration) {
				window.scrollTimeout[id] = setTimeout(animateScroll, increment);
			} else if (endFunction !== false) {
				endFunction();
			}
		};
		animateScroll();
	}
}

export function formatNumber(num, opts = {}) {
	let options = extend({}, {
		showDecimals: false,
		decimals: 2
	}, opts)
	let plainNum = Math.trunc(num);
	let result = plainNum.toString().replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1.');

	if (options.showDecimals) {
		result += (num - plainNum).toFixed(options.decimals).toString().replace('.', ',').replace('0,', ',');
	}

	return result;
}

export function imageLoad(src, returnFunction, additionalData = null) {
	let img = new Image();
	img.onload = function() {
		returnFunction(additionalData);
	}
	img.src = src;
	if (img.complete) {
		img.onload();
	}
}

export function uid(prefix = 'uid') {
	if (!isDefined(window._uidList)) {
		window._uidList = [];
	}
	let id = getid(prefix);

	return id;

	function getid(prefix) {
		let id = prefix + '_' + Math.random().toString(36).substr(2, 9);

		if (window._uidList.includes(id)) {
			id = getid(prefix);
		}
		return id;
	}
}

export function isDefined(variable) {
	return !(typeof variable === 'undefined' || variable === null);
}

export function serializeArray(form, seperator = ',', ignoreEmpty = false) {
	var serialized = {};
	let elements = (form.elements ? form.elements : form.querySelectorAll('input, textarea, select'));
	for (var i = 0; i < elements.length; i++) {

		var field = elements[i];
		if (!field.name || field.disabled || field.type === 'file' || field.type === 'reset' || field.type === 'submit' || field.type === 'button') continue;

		if (field.type === 'select-multiple') {
			for (var n = 0; n < field.options.length; n++) {
				if (!field.options[n].selected) continue;
				let val = field.options[n].value;

				if (!(ignoreEmpty && val === '')) {
					let name = field.name.replace('[]', '');
					if (serialized[name] && name + '[]' === field.name) {
						val = serialized[name] += seperator + val;
					}
					serialized[name] = val;
				}

				//serialized.push(encodeURIComponent(field.name) + "=" + encodeURIComponent(field.options[n].value));
			}
		} else if ((field.type !== 'checkbox' && field.type !== 'radio') || field.checked) {
			let val = field.value;

			if (!(ignoreEmpty && val === '')) {
				let name = field.name.replace('[]', '');
				if (serialized[name] && name + '[]' === field.name) {
					val = serialized[name] += seperator + val;
				}
				serialized[name] = val;
			}
			//serialized.push(encodeURIComponent(field.name) + "=" + encodeURIComponent(field.value));
		}
	}

	return serialized;
};

export function serialize(form, seperator = ',', ignoreEmpty = false) {
	var data = serializeArray(form, seperator, ignoreEmpty);
	var serialized = [];

	for (let k = 0; k < Object.keys(data).length; k++) {
		let key = Object.keys(data)[k];
		serialized.push(encodeURIComponent(key) + '=' + encodeURIComponent(data[key]));
	}
	return serialized.join('&');
}

export function blockOverflow(block = true) {
	if (block) {
		let scrollBarWidth = window.innerWidth - document.documentElement.clientWidth;
		document.documentElement.style.marginRight = scrollBarWidth + 'px'
		document.body.classList.add('block-overflow');
		disableBodyScroll(document.querySelector('.filters-content *'));
	} else {
		document.documentElement.style.marginRight = ''
		document.body.classList.remove('block-overflow');
		enableBodyScroll(document.querySelector('.filters-content *'));
	}
}

export function loadScript(url, endFunction) {
	var script = document.createElement("script");
	script.src = url;
	if (isDefined(endFunction())) {
		script.onload = function() {
			endFunction();
		};
	}
	document.getElementsByTagName("body")[0].appendChild(script);
}

export function isExact(object1, object2) {
	return JSON.stringify(object1) === JSON.stringify(object2);
}

export function storageSpace(folder, file) {
	if (file === '' || file === null) {
		return false;
	}
	if (storagePath === 'https://res.cloudinary.com/minoto/image/upload/') {
		return `${storagePath}${folder}/${file}${file.substring(file.length - 4)}`;
	}
	return `${storagePath}${folder}/${file}`;
}

export function apiPath(endpoint) {
	return `${apiBase}${endpoint}`;
}

export function nl2br(str, is_xhtml) {
	var breakTag = (is_xhtml || typeof is_xhtml === 'undefined') ? '<br />' : '<br>';
	return (str + '').replace(/([^>\r\n]?)(\r\n|\n\r|\r|\n)/g, '$1' + breakTag + '$2');
}

export function remToPx(num) {
	return num * parseFloat(getComputedStyle(document.documentElement).fontSize);
}

export function pxToRem(num) {
	return num / parseFloat(getComputedStyle(document.documentElement).fontSize);
}

export function setCookie(cname, cvalue, exdays) {
	var d = new Date();
	d.setTime(d.getTime() + (exdays * 24 * 60 * 60 * 1000));
	var expires = "expires=" + d.toUTCString();
	document.cookie = cname + "=" + cvalue + ";" + expires + ";path=/";
}

export function getCookie(cname) {
	var name = cname + "=";
	var ca = document.cookie.split(';');
	for (var i = 0; i < ca.length; i++) {
		var c = ca[i];
		while (c.charAt(0) === ' ') {
			c = c.substring(1);
		}
		if (c.indexOf(name) === 0) {
			return c.substring(name.length, c.length);
		}
	}
	return false;
}

Math.linearTween = function(t, b, c, d) {
	return c * t / d + b;
};

Math.easeInOutQuad = function(t, b, c, d) {
	t /= d / 2;
	if (t < 1) return c / 2 * t * t + b;
	t--;
	return -c / 2 * (t * (t - 2) - 1) + b;
};