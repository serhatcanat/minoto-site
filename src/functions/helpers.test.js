import {apiPath, formatNumber, isDefined, seoFriendlyUrl, storageSpace} from "./helpers";
import {apiBase, storagePath} from "../config";
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from "enzyme";

configure({adapter: new Adapter()});

describe('is number formatted', () => {

    test('Show Decimal == False', () => {
        expect(formatNumber(255000,)).toBe('255.000');
    });

    test('Show Decimal == True', () => {
        expect(formatNumber(255000, {showDecimals: true})).toBe('255.000,00');
    });
});

describe('is defined', () => {
    test('Is Defined == True', () => {
        let test = 1;
        expect(isDefined(test)).toBe(true);
    });
    test('Show Decimal == False', () => {
        let test;
        expect(isDefined(test)).toBe(false);
    });
});

test('Is url seo friendly', () => {
    expect(seoFriendlyUrl('url to be this ü İ')).toBe('url-to-be-this-u-i');
});

describe('Storage space', () => {

    test('Storage Space folder with valid parameters', () => {
        expect(storageSpace('images', 'media.png')).toBe(`${storagePath}images/media.png.png`);
    });

    const faildeOptions = [null,''];

    faildeOptions.forEach(function (value) {
        test(`Storage Space folder file == ${value}`, () => {
            expect(storageSpace('media.png',value)).toBe(false);
        });
    });


});

test('Is api path valid', () => {
    expect(apiPath('samplePath')).toBe(`${apiBase}samplePath`);
});


export class TestForm extends React.Component {
    constructor(props) {
        super(props)
        this.onTest = this.onTest.bind(this);
    }

    onTest(e){
        console.log('test');
        console.log(e)
    }

    render() {

        return (
            <form onSubmit={this.onTest}>
                <input name='dname' value='test val'/>
                <input name='dname1' value='test val 1'/>
                <button type='submit'>Test Btn</button>
            </form>
        )
    }
}


describe('Form Is serialized', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<TestForm/>);
    });

    it('renders three  components', () => {
        const form = wrapper.find('form');
        form.find('form')
            .simulate('submit', { preventDefault () {} });
    });

});
