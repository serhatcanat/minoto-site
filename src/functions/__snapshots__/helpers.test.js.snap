// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Is serialized renders three components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TestForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="dname"
          value="test val"
        />,
        <input
          name="dname1"
          value="test val 1"
        />,
        <button
          formAction="submit"
        >
          Submit Btn
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "dname",
          "value": "test val",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "dname1",
          "value": "test val 1",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit Btn",
          "formAction": "submit",
        },
        "ref": null,
        "rendered": "Submit Btn",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="dname"
            value="test val"
          />,
          <input
            name="dname1"
            value="test val 1"
          />,
          <button
            formAction="submit"
          >
            Submit Btn
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "dname",
            "value": "test val",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "dname1",
            "value": "test val 1",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit Btn",
            "formAction": "submit",
          },
          "ref": null,
          "rendered": "Submit Btn",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
